src/target/at32f43x.c

```diff
@@ -304,6 +324,8 @@ static bool at32f43_flash_erase(target_flash_s *target_flash, target_addr_t addr
 	}
 
 	at32f43_flash_clear_eop(target, bank_reg_offset);
+	DEBUG_TARGET("%s: 0x%08" PRIX32 "+%" PRIu32 " reg_base 0x%08" PRIX32 "\n", __func__, addr, (uint32_t)len,
+		bank_reg_offset + AT32F43x_FLASH_REG_BASE);
```

Comment:
This and the change in the Flash write routine are unrelated to the option bytes support, so should perhaps be dropped from this PR. A version of them does seem potentially very useful though.

Reply:
Assume I forgot to add this quality of life feature back when I wrote the driver. stm32f1.c has it, as does stm32h7.c fwiw (the two familiar to me drivers). If I drop this change, then either it makes for its own small PR, or gets postponed to the flash_reg_base refactor. I chose lesser of evils. Still, it's in its own separate commit. Should I add similar log statements to option_write_erased/option_erase?

Reply:
Oh, that's fair then. Ok, we'll allow that despite being technically unrelated. Not sure whether to bother in the option bytes side of things.. though, given how option bytes is usually the sorest spot in the STM32 support implementations, perhaps that is a good idea to do.

If you think you can add reasonable and useful logging into the option bytes erase and write routines, then yes, please do so.

Reply:
Added logging. Detected always full erases. Noticed a bug on line 444, I added an extra offset in fast path, making it never succeed and go slow path.